gms/application_state.hh

```diff
@@ -65,8 +65,8 @@ enum class application_state {
     SHARD_COUNT,
     IGNORE_MSB_BITS,
     CDC_STREAMS_TIMESTAMP,
+    CLUSTER_INFO,
```

Comment:
Why not put SNITCH_NAME and CLUSTER_NAME as separate states?

Reply:
The argument against it (not mine) was that the `application_state` should not grow forever.

Reply:
But you grow it anyway now. What do we gain by packing things into one state instead of separate states?